hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java

```diff
@@ -437,6 +475,32 @@ private void stop(String hdfsUser) throws AppException {
         dfsService.closeDfsClient(dfso);
       }
     }
+
+    try {
+      String experimentsIndex = this.project.getName() + "_experiments";
+      // when jupyter is shutdown the experiment status should be updated accordingly as KILLED
+      for (LivyMsg.Session session : sessions) {
+        String sessionAppId = session.getAppId();
+
+        String experiment = elasticController.findExperiment(experimentsIndex, sessionAppId);
+
+        JSONObject json = new JSONObject(experiment);
+        json = json.getJSONObject("hits");
+        JSONArray hits = json.getJSONArray("hits");
+        for(int i = 0; i < hits.length(); i++) {
+          JSONObject obj = (JSONObject)hits.get(i);
+          JSONObject source = obj.getJSONObject("_source");
+          String status = source.getString("status");
+
+          if(status.equals("RUNNING")) {
```

Comment:
@robzor92 is this the YARN application state? if so, you can use [this](https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/jobhistory/JobState.java#L50) one. the name of the class is misleading but it's the one to use.

Reply:
Yes it is, will do.